src/App.tsx

```diff
@@ -8,55 +8,131 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getUserPosts, getUsers } from './api/client';
+import { formatPosts, formatUsers } from './utils/formatData';
+import { UserResponse } from './types/UserResponse';
+import { Notification } from './components/Notification';
+import { PostResponse } from './types/PostResponse';
+import { NewCommentForm } from './components/NewCommentForm';
+import { Comment } from './types/Comment';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [user, setUser] = useState<User>();
+  const [isPostsLoading, setIsPostsLoading] = useState(false);
+  const [posts, setPosts] = useState<Post[]>();
+  const [selectedPost, setSelectedPost] = useState<Post>();
+  const [isNewCommentForm, setIsNewCommentForm] = useState(false);
+  const [newComment, setNewComment] = useState<Comment | null>(null);
+  const [isPostsError, setIsPostsError] = useState(false);
+  const [isCommentsError, setIsCommentsError] = useState(false);
+
+  useEffect(() => {
+    getUsers()
+      .then((res) => {
+        setUsers(formatUsers(res as UserResponse[]));
+      });
+  }, []);
+
+  useEffect(() => {
+    if (user) {
+      getUserPosts(user.id)
+        .then((res) => {
+          setPosts(formatPosts(res as PostResponse[]));
+        })
+        .catch(() => {
+          setIsPostsError(true);
+        })
+        .finally(() => {
+          setIsPostsLoading(() => false);
```

Comment:
```suggestion
          setIsPostsLoading(() => false);
```
why can't we just pass `false` instead of arrow function?